storm-core/src/jvm/org/apache/storm/metric/util/DataPointPopulator.java

```diff
@@ -0,0 +1,61 @@
+package org.apache.storm.metric.util;
+
+import org.apache.storm.metric.api.IMetricsConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class DataPointPopulator implements Serializable {
```

Comment:
Yes `expand` would be better. I was not sure which word is good to represent the behavior. We can replace all `populate` to `expand`.